doc/tutorials/introduction/transition_guide/transition_guide.markdown

```diff
@@ -0,0 +1,216 @@
+Transition guide {#tutorial_transition_guide}
+================
+
+@tableofcontents
+
+Changes overview {#tutorial_transition_overview}
+================
+This document is intended to software developers who wants to migrate their code to OpenCV 3.0.
+
+OpenCV 3.0 introduced many new algorithms and features compared to version 2.4. Some modules have been rewritten, some have been reorganized. Although most of the algorithms from 2.4 are still present, the interfaces can differ.
+
+This section describes most notable changes in general, all details and examples of transition actions are in the next part of the document.
+
+##### Contrib repository
+<https://github.com/Itseez/opencv_contrib>
+
+This is a place for all new, experimental and non-free algorithms. It does not receive so much attention from the support team comparing to main repository, but the community makes an effort to keep it in a good shape.
+
+To build OpenCV with _contrib_ repository, add the following option to your cmake command: `-DOPENCV_EXTRA_MODULES_PATH=<path-to_opencv_contrib>/modules`
+
+##### Headers layout
+Old headers were like `opencv2/<module>/<module>.hpp` and now there are top-level module headers containing the most of the functionality: `opencv2/<module>.hpp`. All C-style API were moved to separate headers like `opencv2/core/core_c.h`.
+
+##### Algorithm interfaces
+General algorithm usage pattern has changed: now it must be created on heap wrapped in smart pointer `cv::Ptr`. Version 2.4 allowed both stack and heap allocations, directly or via smart pointer.
```

Comment:
I would add an explicit sample here. The Q&A has been flooded with questions about this!